### PR TITLE
MSM/Serialization: implement third constraint

### PR DESCRIPTION
**Pull request by dannywillems**

### CODE REVIEW

File: msm/src/serialization/witness.rs [hunk 1]
```diff
@@ -274,17 +278,49 @@ pub fn deserialize_field_element<Fp: Field, Env: InterpreterEnv<Fp>>(
...
+    let c15_var = {
+        let c15 = Env::get_column_for_msm_limb(15);
+        env.bitmask_be(&input_limb2, 64, 49, c15)
+    };
+    let c16_var = {
+        let c16 = Env::get_column_for_msm_limb(16);
+        env.bitmask_be(&input_limb2, 79, 64, c16)
+    };
+
+    // Unfolding for readability.
```
Comment (querolita) [inline msm/src/serialization/witness.rs:306]: makes it less readable to me 😅

File: msm/src/serialization/witness.rs [hunk 2]
```diff
@@ -274,17 +278,49 @@ pub fn deserialize_field_element<Fp: Field, Env: InterpreterEnv<Fp>>(
...
+
+    constraint = constraint + c13_var * cst.clone();
+    cst = cst * Env::constant(shl_15);
+    constraint = constraint + c14_var * cst.clone();
+    cst = cst * Env::constant(shl_15);
+    constraint = constraint + c15_var * cst.clone();
+    cst = cst * Env::constant(shl_15);
+    constraint = constraint + c16_var * cst.clone();
+
+    constraint = (1..20).fold(constraint, |constraint, i| {
```
Comment (querolita) [inline msm/src/serialization/witness.rs:319]: you probably want to comment the equation in the https://github.com/o1-labs/proof-systems/tree/master/msm/src/serialization file

File: msm/src/serialization/witness.rs (hunk 1 above)
Comment (dannywillems) [reply msm/src/serialization/witness.rs:306]: Ok, let me change it then.

File: msm/src/serialization/witness.rs (hunk 2 above)
Comment (dannywillems) [reply msm/src/serialization/witness.rs:319]: It is, see second equation:
```

2. For the remaining 75 bits:

$$c_{12} + c_{13} * 2^{15} + c_{14} 2^{15 * 2} + c_{15} 2^{15 * 3} + c_{16} 2^{15 * 4} = \sum_{j = 1}^{19} b_{2, j} * 2^{4 (j - 1)}$$
```

There is a wrong index though. It is not 19 but 20. Fixing it in a following PR

Comment (dannywillems) [reply msm/src/serialization/witness.rs:319]: Ah no, it is correct. Starting at 0...

File: msm/src/serialization/witness.rs (hunk 1 above)
Comment (dannywillems) [reply msm/src/serialization/witness.rs:306]: Lemme know if it is better for you: https://github.com/o1-labs/proof-systems/pull/1901

**Pull request merged**